eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1073,6 +1092,36 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
 
     case Event(BITCOIN_FUNDING_TIMEOUT, d: DATA_CLOSING) => handleFundingTimeout(d)
 
+    case Event(w: WatchFundingConfirmedTriggered, d: DATA_CLOSING) =>
+      d.alternativeCommitments.find(_.commitInput.outPoint.txid == w.tx.txid) match {
+        case Some(commitments1) =>
+          // This is a corner case where:
+          //  - we are using dual funding
+          //  - *and* the funding tx was RBF-ed
+          //  - *and* we went to CLOSING before any funding tx got confirmed (probably due to a local or remote error)
+          //  - *and* an older version of the funding tx confirmed and reached min depth (it won't be re-orged out)
+          //
+          // This means that:
+          //  - the whole current commitment tree has been double-spent and can safely be forgotten
+          //  - from now on, we only need to keep track of the commitment associated to the funding tx that got confirmed
+          //
+          // Force-closing is our only option here, if we are in this state the channel was closing and it is too late
+          // to negotiate a mutual close.
+          log.info("an alternative funding tx with txid={} got confirmed", w.tx.txid)
+          watchFundingTx(commitments1)
+          context.system.eventStream.publish(TransactionConfirmed(d.channelId, remoteNodeId, w.tx))
+          val commitTx = commitments1.fullySignedLocalCommitTx(keyManager).tx
+          val localCommitPublished = Closing.LocalClose.claimCommitTxOutputs(keyManager, commitments1, commitTx, nodeParams.currentBlockHeight, nodeParams.onChainFeeConf)
+          val d1 = DATA_CLOSING(commitments1, None, d.waitingSince, alternativeCommitments = Nil, mutualCloseProposed = Nil, localCommitPublished = Some(localCommitPublished))
+          stay() using d1 storing() calling doPublish(localCommitPublished, commitments1)
+        case None if d.commitments.commitInput.outPoint.txid == w.tx.txid =>
+          // The best funding tx candidate has been confirmed, we can forget alternative commitments.
+          stay() using d.copy(alternativeCommitments = Nil)
+        case None =>
```

Comment:
Nit: order-sensitive pattern matching are a bit footgunny, we could do:
```scala
case None => if (...) {} else {}
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2274/commits/f49116ab7af3733bd2b1055c66e8eaa27236276b